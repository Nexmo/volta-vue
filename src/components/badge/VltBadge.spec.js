import { shallowMount } from '@vue/test-utils';
import { VltBadge } from '..';

describe('vlt-badge', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltBadge);
  });

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is red', () => {
    wrapper.setProps({ color: 'red' });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is dismissable', () => {
    wrapper.setProps({ dismissable: true });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is dismissed', async () => {
    wrapper.setProps({ dismissable: true });
    wrapper.find('button').trigger('click');
    expect(wrapper.emitted('dismissed')).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
