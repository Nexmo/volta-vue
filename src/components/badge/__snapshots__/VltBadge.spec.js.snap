// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-badge is dismissable 1`] = `
<span
  class="Vlt-badge Vlt-badge--dismissable Vlt-badge--blue"
>
  
  
  <button
    class="Vlt-badge__dismiss"
  />
</span>
`;

exports[`vlt-badge is dismissed 1`] = `<!--v-if-->`;

exports[`vlt-badge is red 1`] = `
<span
  class="Vlt-badge Vlt-badge--red"
>
  
  
  <!--v-if-->
</span>
`;

exports[`vlt-badge renders correctly 1`] = `
<span
  class="Vlt-badge Vlt-badge--blue"
>
  
  
  <!--v-if-->
</span>
`;
